aws_xray_sdk/core/patcher.py

```diff
@@ -10,6 +10,7 @@
     'requests',
     'sqlite3',
     'mysql',
+    'httplib',  # TODO: perhaps requests should map to this below
```

Comment:
I would keep the requests patch for now. If a user's intention is to only patch requests but has other libraries depend on httplib, then patching at a lower level may result in unexpected behavior. IMO the SDK should only patch httplib when users explicit do so. 

A small paragraph on https://github.com/aws/aws-xray-sdk-python/blob/master/docs/thirdparty.rst explaining httplib/httplib2/requests dependency and which to patch could be very helpful.

Reply:
removed comment, and added some docs